medusa/post_processor.py

```diff
@@ -856,22 +855,37 @@ def _should_process(current_quality, new_quality, allowed, preferred):
         """
         Determine if a quality should be processed according to the quality system.
 
+        This method is used only for replace existing files
+        Despite quality system rules (should_search method), in should_process method:
+         - New higher Allowed replaces current Allowed (overrrides rule where Allowed is final quality)
+         - New higher Preferred replaces current Preferred (overrides rule where Preffered is final quality)
+
         :param current_quality: The current quality of the episode that is being processed
         :param new_quality: The new quality of the episode that is being processed
         :param allowed: Qualities that are allowed
         :param preferred: Qualities that are preferred
-        :return: True if the quality should be processed, False or None otherwise.
+        :return: Tuple with Boolean if the quality should be processed and String with reason if should process or not
         """
+        if current_quality is common.Quality.NONE:
+            return False, 'There is no current quality. Skipping as we can only replace existing qualities'
```

Comment:
Its not useless. It make sure it will run only when there is valid quality. Fail safe.